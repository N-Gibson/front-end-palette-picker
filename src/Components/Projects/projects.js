import React, { Component } from 'react';
import { getProjects, postProject, postPalette, deleteProject, deletePalette, getPalettes } from '../../apiCalls';
import Project from '../Project/project';
import './projects.scss';


class Projects extends Component {
  constructor(props) {
    super(props);
    this.state = {
      projectName: '',
      revisedName: '',
      projects: [],
      error: '',
      palettes: []
    }
  }

  async componentDidMount() {
    try {
      const projects = await getProjects();
      this.setState({ projects: projects });
      const palettes = await getPalettes()
      this.setState({ palettes: palettes})
    } catch (error) {
      this.setState({ error: error })
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleClick = async () => {
    if(this.state.projectName) {
      postProject(this.state.projectName);
      this.setState({ projectName: ''});
    }
  }

  deleteProjectAndPalettes = async (projectId) => {
    await this.state.palettes.forEach(palette => {
      if (palette.projectId === projectId){
        this.removePalette(palette.id)
      }
    })
    await deleteProject(projectId)
    getProjects();
    getPalettes();
  }

  removePalette = async (id) => {
    await deletePalette(parseInt(id))
    getPalettes();
  }

  render() {
  const {handleProject} = this.props
  const projects = this.state.projects.map(project => <Project deleteProjectAndPalettes={this.deleteProjectAndPalettes} key={project.id} id={project.id} name={project.name} handleProject={handleProject} changeName={(e) => this.handleChange(e)}/>);
    return(
      <main>
        <h1>Palette Picker</h1>
        <div className='project-inputs'>
          <input 
            className='project-input' 
            type='text' 
            placeholder='New Project Name' 
            name='projectName' 
            value={this.state.projectName} 
            onChange={this.handleChange}>
          </input>
          <button 
            className='project-button' 
            onClick={this.handleClick}>
            Save
          </button>
        </div>
        <div className='project-display'>
          <p className='project-prompt'>Select a project to view palettes</p>
          {projects}
        </div>
      </main>
    )
  }
}

export default Projects