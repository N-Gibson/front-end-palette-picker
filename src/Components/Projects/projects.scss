main {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 10% 1fr 2fr;
  grid-template-areas:
    'nav'
    'inputs'
    'projects';
  min-height: 100vh;
}

h1 {
  display: flex;
  flex-direction: flex-start;
  font-family: 'Dancing Script';
  grid-area: nav;
  margin-left: 20px;
}

.project-inputs { 
  display: flex;
  flex-direction: column;
  grid-area: inputs;
  justify-content: center;
  margin: 0px auto;
  width: 50%;
}

.project-input {
  border-radius: 20px;
  font-size: 1em;
  height: 50px;
  margin-bottom: 5px;
  padding-left: 10px;
}

.project-button {
  border-radius: 20px;
  cursor: pointer;
  font-size: 1.2em;
  height: 50px;
  margin-top: 5px;
    &:hover {
      background-color: lightgrey;
    }
}

.project__button-interactions {
  border-radius: 20px;
}

.project-display {
  align-items: center;
  background-color: #D9DCDF;
  display: grid;
  grid-area: projects;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  grid-gap: 20px;
  padding: 20px;
    &:hover .project-prompt {
      color: black;
    }
}

.project-prompt {
  color: #D9DCDF;
  left: 42%;
  position: absolute;
}

.project-name {
  border-bottom: 1px solid black;
  padding-bottom: 5px;
}

.project {
  align-items: center;
  border: 1px solid;
  cursor: pointer;
  display: flex;
  height: 15vh;
  justify-content: center;
   &:hover {
     background-color: #fff;
   }
}
