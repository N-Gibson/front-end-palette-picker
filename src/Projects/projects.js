import React, { Component } from 'react';
import { getProjects, postProject } from '../apiCalls';
import Project from '../Project/project';
import './projects.scss';

class Projects extends Component {
  constructor(props) {
    super(props);
    this.state = {
      projectName: '',
      projects: [],
    }
  }

  async componentDidMount() {
    try {
      const projects = await getProjects();
      this.setState({ projects: projects })
    } catch (error) {
      this.setState({ error: error })
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleClick = async () => {
    if(this.state.projectName) {
      postProject(this.state.projectName);
      this.setState({ projectName: ''});
    }
  }

  render() {
  const projects = this.state.projects.map(project => <Project key={project.id} id={project.id} name={project.name} changeName={() => this.handleChange()}/>);
    return(
      <main>
        <h1>Palette Picker</h1>
        <div className='project-inputs'>
          <input 
            className='project-input' 
            type='text' 
            placeholder='New Project Name' 
            name='projectName' 
            value={this.state.projectName} 
            onChange={this.handleChange}>
          </input>
          <button 
            className='project-button' 
            onClick={this.handleClick}>
            Save
          </button>
        </div>
        <div className='project-display'>
          <p className='project-prompt'>Select a project to view palettes</p>
          {projects}
        </div>
      </main>
    )
  }
}

export default Projects