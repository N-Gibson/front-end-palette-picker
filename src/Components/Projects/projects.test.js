import React from 'react';
import { shallow } from 'enzyme';
import Projects from './Projects'

describe('Projects', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Projects />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})