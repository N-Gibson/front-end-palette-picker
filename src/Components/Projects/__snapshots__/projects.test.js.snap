// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot 1`] = `
<main>
  <h1>
    Palette Picker
  </h1>
  <div
    className="project-inputs"
  >
    <input
      className="project-input"
      name="projectName"
      onChange={[Function]}
      placeholder="New Project Name"
      type="text"
      value=""
    />
    <button
      className="project-button"
      onClick={[Function]}
    >
      Save
    </button>
  </div>
  <div
    className="project-display"
  >
    <p
      className="project-prompt"
    >
      Select a project to view palettes
    </p>
  </div>
</main>
`;
